lib/MusicBrainz/Server/Data/Artist.pm

```diff
@@ -465,12 +465,37 @@ sub _hash_to_row
 }
 
 sub load_related_info {
-    my ($self, @artists) = @_;
+    my ($self, $artists_ref, $user_lang) = @_;
 
     my $c = $self->c;
+    my @artists = @{$artists_ref};
     $c->model('ArtistType')->load(@artists);
     $c->model('Gender')->load(@artists);
     $c->model('Area')->load(@artists);
+
+    if (defined $user_lang) {
+        my $short_user_lang = substr($user_lang, 0, 2);
+        my $aliases = $c->model('Artist')->alias->find_by_entity_ids(
+            map { $_->id } @artists,
+        );
+        for my $artist (@artists) {
+            my $best_alias;
+            foreach my $alias (@{$aliases->{$artist->id}}) {
```

Comment:
This logic is essentially duplicated this file and Search.pm, but that may be unavoidable due to the alias objects being different (see e.g. `primary_for_locale` here vs. `primary` in Search.pm).

Reply:
Perhaps we can build `Entity::Alias` objects in Search.pm so that they can be passed to a utility function that can be called from both files?

Technically there are separate alias classes per entity type (e.g., `Entity::ArtistAlias`), but the generic class `Entity::Alias` may suffice for the limited amount information we need to store to satisfy the utility code.

Reply:
This makes sense to me. Can you give me a pointer to some existing code that's manually creating entity objects? I've spun my wheels for a bit now without having any luck here -- my OO Perl knowledge is pretty much nonexistent (like the rest of my Perl knowledge). I think that the only fields that I care about are `locale`, `name`, and `primary_for_locale` (which needs to be remapped from `primary` in Search.pm).

Reply:
Never mind, I think I figured this out. :-)